lib/remote_persistent_term/fetcher/s3.ex

```diff
@@ -149,58 +166,67 @@ defmodule RemotePersistentTerm.Fetcher.S3 do
 
   defp find_latest(_), do: {:error, :not_found}
 
-  defp aws_client_request(op, %{region: region, failover_regions: nil}),
-    do: client().request(op, region: region)
+  defp aws_client_request(op, %{failover_buckets: nil} = state, opts) do
+    perform_request(op, state.bucket, state.region, opts)
+  end
 
   defp aws_client_request(
          op,
          %{
-           region: region,
-           bucket: bucket,
-           key: key,
-           failover_regions: failover_regions
-         } = state
+           failover_buckets: failover_buckets
```

Comment:
```suggestion
           failover_buckets: [_|_] = failover_buckets
```

just to make sure the list isn't empty. We can get rid of the guard with this change too